src/nlp/api.jl

```diff
@@ -184,10 +184,15 @@ function jac_structure!(
   @lencheck nlp.meta.nnzj rows cols
   lin_ind = 1:(nlp.meta.lin_nnzj)
   nlp.meta.nlin > 0 && jac_lin_structure!(nlp, view(rows, lin_ind), view(cols, lin_ind))
+  for i in lin_ind
+    rows[i] += count(<(nlp.meta.lin[rows[i]]), nlp.meta.nln)
```

Comment:
I felt this was too obscure and I made a small benchmark and apparently the following is faster and allocates less (?!). But mostly, I think it is clearer
```suggestion
    rows[i] += count(x < nlp.meta.lin[rows[i]] for x in nlp.meta.nln)
```